python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -390,7 +429,9 @@ def _get_dynamic_ray_remote_args(
         return ray_remote_args
 
     @abstractmethod
-    def _add_bundled_input(self, refs: RefBundle):
+    def _add_bundled_input(
```

Comment:
1. I still prefer putting slices inside RefBundle. Because 1) slices should always be used together with a RefBundle; 2) We need to slices info to make RefBundle.num_rows/size_bytes correct. 
2. type of slices should be `List[Optional[BlockSlice]]`. I.E., some blocks need slicing, some don't.